src/go/rpk/pkg/cli/topic/trim.go

```diff
@@ -0,0 +1,279 @@
+// Copyright 2023 Redpanda Data, Inc.
+//
+// Use of this software is governed by the Business Source License
+// included in the file licenses/BSL.md
+//
+// As of the Change Date specified in that file, in accordance with
+// the Business Source License, use of this software will be governed
+// by the Apache License, Version 2.0
+
+package topic
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/config"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/kafka"
+	"github.com/redpanda-data/redpanda/src/go/rpk/pkg/out"
+	"github.com/spf13/afero"
+	"github.com/spf13/cobra"
+	"github.com/twmb/franz-go/pkg/kadm"
+)
+
+func newTrimPrefixCommand(fs afero.Fs, p *config.Params) *cobra.Command {
+	var (
+		all        bool
+		fromFile   string
+		noConfirm  bool
+		offset     string
+		partitions []int
+	)
+	cmd := &cobra.Command{
+		Use:     "trim-prefix [TOPIC]",
+		Aliases: []string{"trim"},
+		Args:    cobra.MaximumNArgs(1),
+		Short:   "Trim records from topics",
+		Long: `Trim records from topics
+
+This command allows you to trim records from topics, to trim the topics Redpanda
+sets the LogStartOffset for partitions to the requested offset. All segments 
+whose max partition is before the requested offset are deleted, and any records 
+within the segment before the requested offset can no longer be read.
+
+The --offset/-o flag allows you to indicate which index you want to set the 
+partition's low watermark (start offset) to. It can be a single integer value 
+denoting the offset or a timestamp if you prefix the offset with an '@'. You may
+select which partition you want to trim the offset from with the --partition/-p 
+flag.
+
+The --from-file option allows to trim the offsets specified in a text file with
+the following format:
+    [TOPIC] [PARTITION] [OFFSET]
+    [TOPIC] [PARTITION] [OFFSET]
+    ...
+or the equivalent keyed JSON/YAML file.
+
+EXAMPLES
+
+Trim records in 'foo' topic to offset 120 in partition 1
+    rpk topic trim-prefix foo --offset 120 --partition 1
+
+Trim records in all partitions of topic foo previous to an specific timestamp
+    rpk topic trim-prefix foo -o "@1622505600"
+
+Trim records from a JSON file
+    rpk topic trim-prefix --from-file /tmp/to_trim.json
+`,
+		Run: func(cmd *cobra.Command, args []string) {
+			p, err := p.LoadVirtualProfile(fs)
+			out.MaybeDie(err, "unable to load config: %v", err)
+
+			adm, err := kafka.NewAdmin(fs, p)
+			out.MaybeDie(err, "unable to initialize kafka client: %v", err)
+			defer adm.Close()
+
+			var o kadm.Offsets
+			if fromFile != "" {
+				o, err = parseOffsetFile(fs, fromFile)
```

Comment:
(see below -- here we would pass the first arg if it exists, the topic, to this function)